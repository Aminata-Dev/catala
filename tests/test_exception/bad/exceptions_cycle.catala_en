## Test

```catala
declaration scope A:
  context x content integer

scope A:
  label base_x
  exception exception_exception_x
  definition x equals 0

  label exception_x
  exception base_x
  definition x equals 1

  label exception_exception_x
  exception exception_x
  definition x equals 2
```

```catala-test-inline
$ catala Interpret -s A
[ERROR] Exception cycle detected when defining x: each of these 3 exceptions applies over the previous one, and the first applies over the last

┌─⯈ tests/test_exception/bad/exceptions_cycle.catala_en:8.2-10.14:
└──┐
 8 │   label base_x
   │   ‾‾‾‾‾‾‾‾‾‾‾‾
 9 │   exception exception_exception_x
   │   ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
10 │   definition x equals 0
   │   ‾‾‾‾‾‾‾‾‾‾‾‾


┌─⯈ tests/test_exception/bad/exceptions_cycle.catala_en:12.2-14.14:
└──┐
12 │   label exception_x
   │   ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
13 │   exception base_x
   │   ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
14 │   definition x equals 1
   │   ‾‾‾‾‾‾‾‾‾‾‾‾


┌─⯈ tests/test_exception/bad/exceptions_cycle.catala_en:16.2-18.14:
└──┐
16 │   label exception_exception_x
   │   ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
17 │   exception exception_x
   │   ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
18 │   definition x equals 2
   │   ‾‾‾‾‾‾‾‾‾‾‾‾

#return code 255#
```
