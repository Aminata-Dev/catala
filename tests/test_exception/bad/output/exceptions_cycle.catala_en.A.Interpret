[ERROR] Cyclic dependency detected between exceptions!

Cyclic exception for definition of variable "x", declared here:
  --> tests/test_exception/bad/exceptions_cycle.catala_en
   | 
16 |   label exception_exception_x
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^
17 |   exception exception_x
   |   ^^^^^^^^^^^^^^^^^^^^^
18 |   definition x equals 2
   |   ^^^^^^^^^^^^^^^^^^^^^^
   + 

Used here in the definition of another cyclic exception for defining "x":
  --> tests/test_exception/bad/exceptions_cycle.catala_en
   | 
18 |   definition x equals 2
   |              ^
   + Test

Cyclic exception for definition of variable "x", declared here:
  --> tests/test_exception/bad/exceptions_cycle.catala_en
   | 
12 |   label exception_x
   |   ^^^^^^^^^^^^^^^^
13 |   exception base_x
   |   ^^^^^^^^^^^^^^^^
14 |   definition x equals 1
   |   ^^^^^^^^^^^^^^^^^^^^^^
   + 

Used here in the definition of another cyclic exception for defining "x":
  --> tests/test_exception/bad/exceptions_cycle.catala_en
   | 
14 |   definition x equals 1
   |              ^
   + Test

Cyclic exception for definition of variable "x", declared here:
  --> tests/test_exception/bad/exceptions_cycle.catala_en
   | 
 8 |   label base_x
   |   ^^^^^^^^^^^
 9 |   exception exception_exception_x
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
10 |   definition x equals 0
   |   ^^^^^^^^^^^^^^^^^^^^^^
   + 

Used here in the definition of another cyclic exception for defining "x":
  --> tests/test_exception/bad/exceptions_cycle.catala_en
   | 
10 |   definition x equals 0
   |              ^
   + Test
