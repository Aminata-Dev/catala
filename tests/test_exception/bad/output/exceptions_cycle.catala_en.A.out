[ERROR] Cyclic dependency detected between exceptions!

Cycle exception exception_exception_x, declared:
  --> test_exception/bad/exceptions_cycle.catala_en
   | 
16 |   label exception_exception_x
   |         ^^^^^^^^^^^^^^^^^^^^^
   + Test

Used here in the definition of another cycle exception exception_x:
  --> test_exception/bad/exceptions_cycle.catala_en
   | 
17 |   exception exception_x
   |             ^^^^^^^^^^^
   + Test

Cycle exception exception_x, declared:
  --> test_exception/bad/exceptions_cycle.catala_en
   | 
12 |   label exception_x
   |         ^^^^^^^^^^^
   + Test

Used here in the definition of another cycle exception base_x:
  --> test_exception/bad/exceptions_cycle.catala_en
   | 
13 |   exception base_x
   |             ^^^^^^
   + Test

Cycle exception base_x, declared:
 --> test_exception/bad/exceptions_cycle.catala_en
  | 
8 |   label base_x
  |         ^^^^^^
  + Test

Used here in the definition of another cycle exception exception_exception_x:
 --> test_exception/bad/exceptions_cycle.catala_en
  | 
9 |   exception exception_exception_x
  |             ^^^^^^^^^^^^^^^^^^^^^
  + Test
